### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by ashrafhasson**
updated with documentation on `infinityModelLoaded` and `infinityModelUpdated`

### CODE REVIEW

**Comment by hhff** [comment]
Hi @ashrafhasson - can you please pull master & rebase against the latest - and squash your PR?  

There should not be any changes to the codebase here - it looks like you've been working against an older version of the codebase.

Also will add some notes to the README changes.

Thankyou so much!!

File: README.md [hunk 1]
```diff
@@ -135,6 +157,33 @@ setupController: function(controller, model) {
   controller.set('products', model);
 }
 ```
+### event hooks
```
Comment (hhff) [inline README.md]: Can we please capitalize this section?

File: README.md [hunk 2]
```diff
@@ -135,6 +157,33 @@ setupController: function(controller, model) {
...
+
+ember-infinity provides a route mixin that enables us to hook into the following events:
```
Comment (hhff) [inline README.md]: It's implied the user is already using the route mixin, we can just say:

"The route mixin also provides the following event hooks:"

File: README.md [hunk 3]
```diff
@@ -135,6 +157,33 @@ setupController: function(controller, model) {
...
+
+* **infinityModelUpdated**
+
+This gets fired up when data is loaded into the model.
```
Comment (hhff) [inline README.md]: I think the word "fired up" is a little ambigious.  Can we change this to:

---

Triggered on the route whenever new objects are pushed into the infinityModel.

Args:
- totalPages

File: README.md [hunk 4]
```diff
@@ -135,6 +157,33 @@ setupController: function(controller, model) {
...
+
+* **infinityModelLoaded**
+
+Fired up when there's no more data to load from the backend, needless to say this depends on your `meta.total_pages` key returned by your server.
```
Comment (hhff) [inline README.md]: Same here, can we change this to:

Triggered on the route when the infinityModel is fully loaded.

**Args:**
- lastPageLoaded
- totalPages
- infinityModel

File: README.md [hunk 5]
```diff
@@ -135,6 +157,33 @@ setupController: function(controller, model) {
...
+export default Ember.Route.extend(InfinityRoute, {
+  ...
+
+	infinityModelUpdated: function() {
+		Ember.Logger.debug('updated with more items');
+	},
+	infinityModelLoaded: function() {
+		Ember.Logger.info('no more items to load');
+	}
+}
```
Comment (hhff) [inline README.md]: would be cool to include the args here also :heart:

**Comment by ashrafhasson** [comment]
Hi @hhff, thanks for taking the time to look into my commit. I'll implement your suggestions shortly. I just wanted to confirm that I've initially branched off of 5fdd869, this seems to be your latest commit, no?
Also, I've pushed two more commits on top of that, should I just push the new changes I make on top of those and you'll pick what you deem necessary or do you want a new clean branch for the documentation?

**Comment by ashrafhasson** [comment]
@hhff please let me know if the following is good for merging: ashrafhasson@7a99844fcd45ce08b6499f75bf5873db695aade0

File: README.md (hunk 5 above)
Comment (hhff) [inline README.md]: Can we please move to soft tabs (2 spaces) and show the arguments in these examples?

File: README.md (hunk 3 above)
Comment (hhff) [inline README.md]: when you read this in Markdown - it looks like **infinityModelUpdated** is an argument for **infinityModelLoaded**

can we please remove the * from the function names so it is not rendered as an <li> tag?

**Comment by hhff** [comment]
@ashrafhasson added a couple more comments to the README.

Yes, please remove anything that is not associated with the README from this PR.  1x PR per feature please!

I love that we're supporting computed properties on the route!  It will just need to be a separate PR.

File: README.md (hunk 5 above)
Comment (ashrafhasson) [inline README.md]: sure thing, I'll make the changes but just for clarity, you want to show the arguments in the function definition as in: 

```js
infinityModelUpdated: function(totalPages) { ... }
```

and 

```js
infinityModelLoaded: function(lastPageLoaded, totalPages, infinityModel) { ... }
```

?

Comment (hhff) [inline README.md]: Yup!

**Comment by ashrafhasson** [comment]
closing this, an alternative PR was merged in 5512bd70ae2ddbf48aeeaa941a4cc7c047f2f14a

**Pull request closed without merging**